### PR TITLE
Tmedia 751 card list refactor v2

### PR DESCRIPTION
**Pull request by JackHowa**
## Description

Convert the card list to the v2 architecture

Based off of https://github.com/WPMedia/arc-themes-blocks/pull/1393

## Jira Ticket

- [TMEDIA-751](https://arcpublishing.atlassian.net/browse/TMEDIA-751)

## Acceptance Criteria

Card List block is migrated to the new architecture.

### Chromatic 
- Compare with storybook via the chromatic compare diffs 
- You will notice that the font changed 
- And that the header spacing is slightly different as noted in the discrepancies now matching the designs

### Live 

1. Make sure your env is pointing to themes sandbox internal
2. Checkout this branch `git checkout TMEDIA-751-card-list` of feature pack
3. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/card-list-block`
4. Create a new page 
5. Add the card list block 
6. Update the title to be Card List to see that set 
7. Set the content to be 
Screen Shot 2022-06-30 at 10 52 13

8. See 
Screen Shot 2022-06-30 at 10 52 49
Screen Shot 2022-06-30 at 10 54 19
9. Change the offset to be 0. See the fallback image set 
Screen Shot 2022-06-30 at 10 57 54

### Before

Some spacing differences as noted in discrepancies doc 

### After

Matches mocks in v2
- image discrepancies match the 40vh of before
- Byline and date font size match v2 and noted in discrepancies

## Dependencies or Side Effects

- Feature pack https://github.com/WPMedia/arc-themes-feature-pack/pull/289
- Manifest update https://github.com/WPMedia/arc-themes-manifests/pull/91

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress. -> ty @nschubach 
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added. -> ty @RobinGiannattasio ! 

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/card-list-block/features/card-list/default.jsx [hunk 1]
```diff
@@ -4,10 +4,25 @@ import PropTypes from "@arc-fusion/prop-types";
 import { useContent } from "fusion:content";
 import { useFusionContext } from "fusion:context";
 import getProperties from "fusion:properties";
-import { Image, LazyLoad, isServerSide } from "@wpmedia/engine-theme-sdk";
+import getTranslatedPhrases from "fusion:intl";
+import { LazyLoad, localizeDate } from "@wpmedia/engine-theme-sdk";
 import { extractResizedParams, extractImageFromStory } from "@wpmedia/resizer-image-block";
```
Comment (matthewroach) [inline blocks/card-list-block/features/card-list/default.jsx]: We need to remove this block from being imported

**Review by matthewroach — CHANGES_REQUESTED** [review]

File: blocks/card-list-block/features/card-list/default.jsx (hunk 1 above)
Comment (JackHowa) [reply blocks/card-list-block/features/card-list/default.jsx]: oh my bad. will work on this

Comment (JackHowa) [reply blocks/card-list-block/features/card-list/default.jsx]: updated -- missed this in one of my stashes

File: blocks/card-list-block/features/card-list/default.jsx [hunk 2]
```diff
@@ -18,7 +33,7 @@ const getFallbackImageURL = ({ deployment, contextPath, fallbackImage }) => {
 
 	return targetFallbackImage;
 };
-
+// f
```
Comment (matthewroach) [inline blocks/card-list-block/features/card-list/default.jsx]:
```suggestion

```

Comment (JackHowa) [reply blocks/card-list-block/features/card-list/default.jsx]: updated -- was using for testing watch

File: blocks/card-list-block/features/card-list/default.jsx [hunk 3]
```diff
@@ -108,95 +132,130 @@ const CardListItems = (props) => {
...
 	return contentItems.length > 0 ? (
 		<HeadingSection>
-			<div className="card-list-container">
-				{title ? <Heading className="card-list-title">{title}</Heading> : null}
-				<Wrapper>
-					<article
-						className="list-item-simple"
-						key={`card-list-${contentItems[0].websites[arcSite].website_url}`}
+			<Stack className={BLOCK_CLASS_NAME}>
+				{title ? <Heading className={`${BLOCK_CLASS_NAME}__title`}>{title}</Heading> : null}
```
Comment (matthewroach) [inline blocks/card-list-block/features/card-list/default.jsx:171]: Looks like the conditional `Wrapper` logic has been removed - if there is a title we need another `HeadingSection` to increment the heading levels below

Comment (JackHowa) [reply blocks/card-list-block/features/card-list/default.jsx:171]: oh damn -- will add back

File: blocks/card-list-block/features/card-list/default.jsx [hunk 4]
```diff
@@ -4,10 +4,25 @@ import PropTypes from "@arc-fusion/prop-types";
...
-import { extractResizedParams, extractImageFromStory } from "@wpmedia/resizer-image-block";
-import { Byline, Heading, HeadingSection, Overline, PromoDate } from "@wpmedia/shared-styles";
-import "./card-list.scss";
+import getTranslatedPhrases from "fusion:intl";
+import { LazyLoad, localizeDate } from "@wpmedia/engine-theme-sdk";
+import {
+	Attribution,
+	Date,
+	formatAuthors,
+	formatUrl,
```
Comment (matthewroach) [inline blocks/card-list-block/features/card-list/default.jsx]:
```suggestion
	formatURL,
```

File: blocks/card-list-block/features/card-list/default.jsx [hunk 5]
```diff
@@ -108,95 +131,130 @@ const CardListItems = (props) => {
...
-								{contentItems[0].headlines.basic}
-							</a>
-						</Heading>
-						<div className="author-date">
-							<Byline content={contentItems[0]} list separator={showSeparator} font="Primary" />
-							<PromoDate className="story-date" date={contentItems[0].display_date} />
-						</div>
-					</article>
+						<Stack className={`${BLOCK_CLASS_NAME}__main-item-text-container`}>
+							{url || text ? <Overline href={url ? formatUrl(url) : null}>{text}</Overline> : null}
```
Comment (matthewroach) [inline blocks/card-list-block/features/card-list/default.jsx]:
```suggestion
							{url || text ? <Overline href={url ? formatURL(url) : null}>{text}</Overline> : null}
```

File: blocks/card-list-block/features/card-list/default.jsx (hunk 3 above)
Comment (JackHowa) [reply blocks/card-list-block/features/card-list/default.jsx:171]: updated

File: blocks/card-list-block/features/card-list/default.jsx (hunk 5 above)
Comment (JackHowa) [reply blocks/card-list-block/features/card-list/default.jsx]: not sure how this got in

**Review by matthewroach — APPROVED** [review]
Looking good from a development perspective - @haganka can you review the chromatic tests?

**Review by matthewroach — APPROVED** [review]

**Comment by JackHowa** [comment]
@haganka updated -- thank you!

**Comment by haganka** [comment]
@JackHowa approved knowing the mobile text style font sizes will be updated separately. Per chat, here is that[ ticket](https://arcpublishing.atlassian.net/browse/TBRANDS-88)

**Pull request merged**